test/test_char_array.cpp

```diff
@@ -122,11 +124,21 @@ TEST_F(ArrayCharTest, vsprintf_fail) {
   ASSERT_EQ(RCUTILS_RET_OK, ret);
 
   char_array.allocator = failing_allocator;
-  ret = example_logger(&char_array, "Long string for the case %d", 2);
+  ret = example_logger(&char_array, "Long string for given %s", "array");
```

Comment:
If this change is important, I would add a comment that describes why a %s is needed instead of a %d

Reply:
Not important, a leftover.